t/misc.t

```diff
@@ -159,3 +159,45 @@ GET /t
  bad argument
 --- error_log
 unsupported subsystem: http
+
+
+
+=== TEST 6: not internal request
+--- config
+    location /test {
+        rewrite ^/test$ /lua last;
+    }
+    location /lua {
+        content_by_lua '
+            if ngx.req.is_internal() then
+                ngx.say("internal")
+            else
+                ngx.say("not internal")
+            end
+        ';
+    }
+--- request
+GET /lua
+--- response_body
+not internal
+
+
+
+=== TEST 7: internal request
+--- config
+    location /test {
+        rewrite ^/test$ /lua last;
+    }
+    location /lua {
+        content_by_lua '
+            if ngx.req.is_internal() then
+                ngx.say("internal")
+            else
+                ngx.say("not internal")
+            end
+        ';
+    }
+--- request
+GET /test
+--- response_body
+internal
```

Comment:
Could you add another test case for FFI_BAD_CONTEXT branch?

Reply:
It seems no `FFI_BAD_CONTEXT` test case for `is_subreq` and `headers_sent`, are there any example?

Reply:
@chronolaw 
You could run `is_internal` in phase doesn't support it, and use `pcall` to capture the result. The result could be returned as the response for `GET /xxx`.

Reply:
It works in all phases except `init_by_lua` and 'init_worker_by_lua' and 'ngx.timer'，should I use `ngx.errorlog` to capture errorlog?

And I thin it will generate ''no request found" but not "API disabled in the current context".

Reply:
> should I use ngx.errorlog to capture errorlog

You could store the result in `package.loaded` or just a variable which could be accessed by the `GET /xxx` request and the `ngx.timer`.

Reply:
It use the `error` function like below
```
error("no request found")
```
This will case a stack traceback in errlog, we can not get any return value.

Reply:
@chronolaw 
Just forget the errlog and use `pcall` to capture the `error` message.